parlai/core/torch_classifier_agent.py

```diff
@@ -163,11 +163,12 @@ def __init__(self, opt: Opt, shared=None):
                 print('Loading existing model parameters from ' + init_model)
                 self.load(init_model)
             if self.use_cuda:
-                self.model.cuda()
+                if self.model_parallel:
```

Comment:
Since this code seems to now be repeated in TCA, TGA, and TRA, why not move to a new method in TA?

Reply:
Yeah, I do think there's a refactor incoming. I've wanted to do it for a while, but subtle differences kept me from being confident in it. I'd like a lot less custom init code in each of those classes...